pkg/azureutils/azure_disk_utils.go

```diff
@@ -194,6 +185,44 @@ func GetCloudProvider(kubeconfig, secretName, secretNamespace, userAgent string)
 	return az, nil
 }
 
+// GetCloudProviderFromConfig get Azure Cloud Provider
+func GetCloudProvider(kubeConfig, secretName, secretNamespace, userAgent string) (*azure.Cloud, error) {
+	kubeClient, err := GetKubeClient(kubeConfig)
+	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
+		return nil, fmt.Errorf("failed to get KubeClient: %v", err)
+	}
+	az, err := GetCloudProviderFromClient(kubeClient, secretName, secretNamespace, userAgent)
+	return az, err
+}
+
+// GetKubeConfig gets config object from config file
+func GetKubeConfig(kubeconfig string) (config *rest.Config, err error) {
+	// attempt to read the in-cluster config
+	config, err = rest.InClusterConfig()
+	// if we couldn't get the in-cluster config
+	// get kubeconfig path from environment variable
+	if err != nil {
+		if kubeconfig == "" {
+			kubeconfig = os.Getenv("KUBECONFIG")
+			if kubeconfig == "" {
+				kubeconfig = filepath.Join(os.Getenv("HOME"), ".kube", "config")
```

Comment:
`HOME` path is actually inside driver container, would v2 driver use `~/.kube/config` as kubeconfig? I doubt that.

Reply:
@mukhoakash I think this was originally your change. I have kept it as an additional fall back scenario but was there any other reason for adding this?

Reply:
I suspect this was done to ease running the driver locally on a dev box since this would pick up whatever the current Kubernetes cluster context had been set to.

Reply:
that would require running this driver as a binary instead of container, that's very rare case, if really needed, just set `kubeconfig` env var. I would suggest keep original logic:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/36b2a9a078fb25728bb909045926e6abfa45565a/pkg/azureutils/azure_disk_utils.go#L445-L461

Reply:
> @mukhoakash I think this was originally your change. I have kept it as an additional fall back scenario but was there any other reason for adding this?

Yes @landreasyan it was kept for easing the driver run from dev box env which @edreed mentioned.